frontend/view-edit-client/interactions/add-client-interaction.component.tsx

```diff
@@ -39,6 +35,15 @@ export default function AddClientInteraction(props: AddClientInteractionProps) {
     ? clientSearchRef.current.clientId
     : props.clientId;
   const userMode = React.useContext(UserModeContext);
+  const [interactionRefs, setInteractionRefs] = React.useState([]);
+
+  React.useEffect(() => {
+    setInteractionRefs((oldRefs) =>
+      Array(tempInteractionIds.length)
+        .fill(null)
+        .map((_, i) => oldRefs[i] || createRef())
```

Comment:
Is this another method of creating dynamic refs?

Reply:
yes